packages/ra-ui-materialui/src/form/SimpleFormIterator.js

```diff
@@ -120,6 +122,63 @@ const SimpleFormIterator = props => {
         fields.push(undefined);
     };
 
+    // check if a custom add button was passed as a prop
+    // If yes, then clone that add the addField() method before its onClick event
+    // If no, then just use the original add button
+    const getAddButton = () => {
+        if (!addButton) {
+            return (
+                <Button
+                    className={classNames('button-add', `button-add-${source}`)}
+                    size="small"
+                    onClick={addField}
+                >
+                    <AddIcon className={classes.leftIcon} />
+                    {translate('ra.action.add')}
+                </Button>
```

Comment:
It would be better to extract this button (and the other one) outside the `SimpleFormIterator`. Then you could just set the default values for `addButton` and `removeButton`. This way you would handle default and custom buttons the same way.

Reply:
We can do 2 things here
1. Create `addButton` & `removeButton` components in the same file as of `SimpleFormIterator`.
2. Or we can move this to two separate files addButton.js & removeButton.js

What are your views on this?

Reply:
I would put them in the same file as they should probably not be exposed. ping @fzaninotto

Reply:
@fzaninotto What do you say?

Reply:
@djhi 
While implementing this I realized that, if we are going to keep these `<DefaultAddButton />` & `<DefaultRemoveButton />` outside the `SimpleFormIterator` then we also need to pass a few things (` source, removeField/addField, index`) through props. I think this will make things more complex.